setup.sh

```diff
@@ -1,9 +1,5 @@
 #!/usr/bin/env bash
 
-# exit when a command fails instead of blindly blundering forward
-set -e
-# treat unset variables as an error and exit immediately
-set -u
```

Comment:
I think we should keep `set -u` and `set -e` at the top, and then disable them just for the part that we handle errors

e.g. in maybe_install
```
set +e
$3
set -e
```